audits/v3/Audit2.md

```diff
@@ -0,0 +1,80 @@
+# Audit report
+
+| Name       | Information                                 |  
+| :--------: | --------------------                        |  
+| Repository | https://github.com/DePayFi/depay-evm-router |  
+| Checked    | `DePayRouterV3.sol`                         |  
+| Branch     | v3                                          |  
+| Time       | 2024-09-30                                  |  
+| Author     | Nikolai Mikhailov                           |  
+
+# Result
+
+| Severity        | Count | Link                                                                                             |  
+| :-------------: | ----: | ------------------------------------------------------------------------------------------------ |  
+| High            | 0     |                                                                                                  |  
+| Medium          | 1     |                                                                                                  |  
+|                 |       | [M01 - ProtocolAmount is not enforced, allowing users to bypass protocol fees](#m01)             |  
+| Low             | 0     |                                                                                                  |  
+|                 |       |                                                                                                  |  
+| Informational   | 1     |                                                                                                  |  
+|                 |       | [I01 - Inconsistent time units used for deadlines (milliseconds vs seconds)](#i02)               |  
+
+---
+
+<a name="m01"></a>
+
+## M01 - ProtocolAmount is not enforced, allowing users to bypass protocol fees
```

Comment:
In the context of payment validation, the `protocolAmount` (fee) is intended to be enforced during post-payment validation. It is not necessary to enforce this fee at the smart contract level, as the design allows for flexibility in handling payments with or without a `protocolAmount` (fee). This approach is intentional, ensuring that payments can be processed even when no protocol fee is required.